commit 5c234aa5e33ed9037354a4c94d0d3e0350abe5eb
Author: Shirish Pargaonkar <shirishpargaonkar@gmail.com>
Date:   Thu Aug 29 08:35:10 2013 -0500

    cifs: Add a variable specific to NTLMSSP for key exchange.
    
    Add a variable specific to NTLMSSP authentication to determine
    whether to exchange keys during negotiation and authentication phases.
    
    Since session key for smb1 is per smb connection, once a very first
    sesion is established, there is no need for key exchange during
    subsequent session setups. As a result, smb1 session setup code sets this
    variable as false.
    
    Since session key for smb2 and smb3 is per smb connection, we need to
    exchange keys to generate session key for every sesion being established.
    As a result, smb2/3 session setup code sets this variable as true.
    
    Acked-by: Jeff Layton <jlayton@samba.org>
    Signed-off-by: Shirish Pargaonkar <shirishpargaonkar@gmail.com>
    Signed-off-by: Steve French <smfrench@gmail.com>

diff --git a/fs/cifs/smb2pdu.c b/fs/cifs/smb2pdu.c
index f56cf99e027a..3d010766f267 100644
--- a/fs/cifs/smb2pdu.c
+++ b/fs/cifs/smb2pdu.c
@@ -491,6 +491,7 @@ SMB2_sess_setup(const unsigned int xid, struct cifs_ses *ses,
 	ses->ntlmssp = kmalloc(sizeof(struct ntlmssp_auth), GFP_KERNEL);
 	if (!ses->ntlmssp)
 		return -ENOMEM;
+	ses->ntlmssp->sesskey_per_smbsess = true;
 
 	/* FIXME: allow for other auth types besides NTLMSSP (e.g. krb5) */
 	ses->sectype = RawNTLMSSP;